src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/AsyncPipeline.T.cs

```diff
@@ -0,0 +1,27 @@
+#nullable enable
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    public readonly partial struct AsyncPipeline<T> : IEquatable<AsyncPipeline<T>>
+    {
+        private static readonly CancellationToken CanceledToken;
+
+        static AsyncPipeline() => CanceledToken = new(canceled: true);
+
+        private readonly ValueTask<T> valueTask;
+
+        private readonly bool hasCanceled;
```

Comment:
It seems it should be named as `isCanceled` instead of `hasCanceled` (hereinafter).